source/slang/slang-emit-spirv.cpp

```diff
@@ -1430,13 +1436,28 @@ struct SPIRVEmitContext
         case kIROp_StructType:
             {
                 List<IRType*> types;
+                bool isUnsizedArrayContained = false;
                 for (auto field : static_cast<IRStructType*>(inst)->getFields())
```

Comment:
I think I spot another issue when I was trying to do so.
I thought if the unsized array has the size of `kIndeterminateSize`, then the struct containing it should also has size bigger than `kIndeterminateSize`. But it's not the case, because after the inst of unsized array was created, IR builder will add an decorate with type of `kIROp_SizeAndAlignmentDecoration` via:
```
        builder.addDecoration(
            type,
            kIROp_SizeAndAlignmentDecoration,
            builder.getIntValue(intType, (IRIntegerValue)rules->ruleName),
            builder.getIntValue(intType, sizeAndAlignment.size),
            builder.getIntValue(intType, sizeAndAlignment.alignment));
```

the problem is that it truncates the size by only taking the low 32-bit. So the size will be 0.

This will cause the struct containing this unsized array won't account the array size to the its size.